api/handlers_instances.go

```diff
@@ -642,3 +642,36 @@ func (s *server) VolumeAttachHandler(w http.ResponseWriter, r *http.Request) {
 
 	handleResponseOk(w, out)
 }
+
+func (s *server) InstanceProfileDeleteHandler(w http.ResponseWriter, r *http.Request) {
+	w = LogWriter{w}
+	vars := mux.Vars(r)
+	account := s.mapAccountNumber(vars["account"])
+	name := vars["name"]
+
+	policy, err := generatePolicy([]string{"iam:GetInstanceProfile", "iam:ListAttachedRolePolicies", "iam:DetachRolePolicy", "iam:ListRolePolicies", "iam:DeleteRolePolicy", "iam:RemoveRoleFromInstanceProfile", "iam:DeleteRole", "iam:DeleteInstanceProfile"})
+	if err != nil {
+		handleError(w, err)
+		return
+	}
+
+	orch, err := s.newIAMOrchestrator(r.Context(), &sessionParams{
+		role:         fmt.Sprintf("arn:aws:iam::%s:role/%s", account, s.session.RoleName),
+		inlinePolicy: policy,
+		policyArns: []string{
+			"arn:aws:iam::aws:policy/AmazonEC2ReadOnlyAccess",
```

Comment:
I don't think we need this here since we're not doing anything with EC2

Reply:
I have updated this to `arn:aws:iam::aws:policy/IAMReadOnlyAccess`, please review.